javascript/packages/cli/src/actions/setup.ts

```diff
@@ -0,0 +1,235 @@
+    const promises = [];
+
+    const multibar = new cliProgress.MultiBar(
+      {
+        clearOnComplete: false,
+        hideCursor: true,
+        format:
+          decorators.yellow("{bar} - {percentage}%") +
+          " | " +
+          decorators.cyan("Binary name:") +
+          " {filename}",
+      },
+      cliProgress.Presets.shades_grey,
+    );
+
+    for (let binary of binaries) {
+      promises.push(
+        new Promise<void>(async (resolve, reject) => {
+          let result = options[binary];
+          if (!result) {
+            console.log("options", options, "binary", binary);
+            throw new Error("Binary is not defined");
+          }
+          const { url, name } = result;
+
+          if (!url) throw new Error("No url for downloading, was provided");
+
+          const response = await fetch(url);
+
+          if (!response.ok)
+            throw Error(response.status + " " + response.statusText);
+
+          const contentLength = response.headers.get(
+            "content-length",
+          ) as string;
+          let loaded = 0;
+
+          const progressBar = multibar.create(parseInt(contentLength, 10), 0);
+          const reader = response.body?.getReader();
+          const writer = fs.createWriteStream(path.resolve(name));
+
+          while (true) {
+            const read = await reader?.read()!;
+            if (read?.done) {
+              writer.close();
+              resolve();
+              break;
+            }
+
+            loaded += read.value.length;
+            progressBar.increment();
+            progressBar.update(loaded, {
+              filename: name,
+            });
+            writer.write(read.value);
+          }
+        }),
+      );
+    }
+
+    await Promise.all(promises);
+    multibar.stop();
+    console.log(
+      decorators.cyan(
+        `\n\nPlease add the current dir to your $PATH by running the command:\n`,
+      ),
+      decorators.blue(`export PATH=${process.cwd()}:$PATH\n\n`),
+    );
+  } catch (err) {
+    console.log(
+      `\n ${decorators.red("Unexpected error: ")} \t ${decorators.bright(
+        err,
+      )}\n`,
+    );
+  }
+};
+
+// Retrieve the latest release for polkadot
+const latestPolkadotReleaseURL = async (
+  repo: string,
+  name: string,
+): Promise<[string, string]> => {
+  try {
+    const releases = await fetch(
+      `https://api.github.com/repos/paritytech/${repo}/releases`,
+    );
+
+    let obj: any;
+    let tag_name;
+
+    const allReleases = await releases.json();
+    const release = allReleases.find((r: any) => {
+      obj = r?.assets?.find((a: any) => a.name === name);
+      return Boolean(obj);
+    });
+
+    tag_name = release.tag_name;
+
+    if (!tag_name) {
+      throw new Error(
+        "Should never come to this point. Tag_name should never be undefined!",
+      );
+    }
+
+    return [
+      `https://github.com/paritytech/${repo}/releases/download/${tag_name}/${name}`,
```

Comment:
The asset from Github expose the "browser_download_url" property so we can replace it with:
```suggestion
      obj.browser_download_url
```

![Screenshot from 2023-04-05 15-39-59](https://user-images.githubusercontent.com/45130584/230083313-65514dcb-e57a-45a0-8c80-4ed7388ca9ea.png);